lib/OpenQA/WebAPI/Plugin/YAMLRenderer.pm

```diff
@@ -44,33 +44,33 @@ sub register {
         # validation of the schema itself which can be useful for development and testing.
         # Returns an array of errors found during validation or otherwise an empty array.
         validate_yaml => sub {
-            my ($self, $yaml, $validate_schema) = @_;
-
-            # Note: Using the schema filename; slurp'ed text isn't detected as YAML
-            my $schema_yaml = $self->app->home->child('public', 'schema', 'JobTemplate.yaml')->to_string;
-            my $validator   = JSON::Validator->new;
+            my ($self, $yaml, $schema_filename, $validate_schema) = @_;
+            my $validator = JSON::Validator->new;
             my $schema;
             my @errors;
+
             try {
+                die "No valid schema specified\n" unless ($schema_filename // '') =~ /\.yaml$/;
```

Comment:
Maybe you should adjust the regex so nobody can sneak out the designated directory using a relative path.